website/docs/core-library/testing.md

```diff
@@ -18,17 +17,27 @@ Github Actions are used to run the H3 test suite for every commit. A variety of
 Coverage information is collected in Coveralls. Because of the self-contained nature of the H3 library, we seek
 to have as close to 100% code coverage as possible.
 
-| Operating system | Compiler    | Processor architecture | Special notes
-| ---------------- | ----------- | ---------------------- | -------------
-| Linux (Ubuntu)   | Clang       | x64                    | clang-format-9 is used to ensure all code is consistently formatted
-| Linux            | Clang       | x64                    | An additional copy of the job runs with Valgrind
-| Linux            | Clang       | x64                    | An additional copy of the job runs with coverage reporting, and excerising the `H3_PREFIX` mechanism.
-| Linux            | gcc         | x64                    | 
-| Mac OS           | Apple Clang | x64                    | 
-| Windows          | MSVC        | x64                    | Static library
-| Windows          | MSVC        | x86                    | Static library
-| Windows          | MSVC        | x64                    | Dynamic library; testing is not run in this configuration
-| Windows          | MSVC        | x86                    | Dynamic library; testing is not run in this configuration
+| Operating system | Compiler    | Build type     | Processor architecture | Special notes
+| ---------------- | ----------- | -------------- | ---------------------- | -------------
+| Linux (Ubuntu)   | Clang       | Debug, Release | x64                    | clang-format-9 is used to ensure all code is consistently formatted
+| Linux            | Clang       | Debug          | x64                    | An additional copy of the job runs with [Valgrind](https://valgrind.org/)
+| Linux            | Clang       | Debug          | x64                    | An additional copy of the job runs with coverage reporting, and excerising the `H3_PREFIX` mechanism.
+| Linux            | gcc         | Debug, Release | x64                    |
+| Mac OS           | Apple Clang | Debug, Release | x64                    |
+| Windows          | MSVC        | Debug, Release | x64                    | Static library
+| Windows          | MSVC        | Debug, Release | x86                    | Static library
+| Windows          | MSVC        | Debug, Release | x64                    | Dynamic library; testing is not run in this configuration
+| Windows          | MSVC        | Debug, Release | x86                    | Dynamic library; testing is not run in this configuration
+
+## Defensive code
+
+H3 uses preprocessor macros borrowed from [SQLite's testing methodology](https://www.sqlite.org/testing.html) to include defensive code in the library. Defensive code is code that handles error conditions for which there are no known test cases to demonstrate it. The lack of known test cases means that without the macros, the defensive cases could inappropriately reduce coverage metrics, disincentivizing including them.
+
```

Comment:
nit/suggestion: What about presenting the next three paragraphs as an unordered list, introduced by something like

```
The macros will behave differently, depending on the build type being "release", "debug", or "coverage":
```

It is probably a matter of personal choice, but I find it can make documentation a little more scannable.

Reply:
One styling idea:

The macros will behave differently, depending on the library build type being "release", "debug", or "coverage":

- **release builds**: The defensive code is included without modification. These branches are intended to be very simple (usually only `return`ing an error code and possibly `free`ing some resources) and to be visually inspectable.
- **debug builds**: The defensive code is included and `assert` calls are included if the defensive code is invoked. Any unit test or fuzzer which can demonstrate the defensive code is actually reached will trigger a test failure and the developers can be alerted to cover the defensive code in unit tests.
- **coverage builds**: The defensive code is not included by replacing its condition with a constant value. The compiler removes the defensive code and it is not counted in coverage metrics.